File: Java/Tests/jdi-uitest-webtests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/composite/PaginationTests.java
package com.epam.jdi.uitests.testing.unittests.tests.composite;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.web.selenium.elements.composite.Pagination;
import com.epam.web.matcher.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.SIMPLE_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.simpleTablePage;
import static com.epam.jdi.uitests.web.settings.WebSettings.getDriver;
import static com.epam.web.matcher.testng.Assert.assertContains;

/**
 * Created by Dmitry_Lebedev1 on 21/12/2015.
 */
public class PaginationTests extends InitTests {
    private Pagination pagination() { return  simpleTablePage.pagination; }

    public static final String[] pages = {"index.html", "contacts.html", "support.html", "dates.html",
            "complex-table.html", "simple-table.html", "user-table.html", "table-pages.html",
            "different-elements.html", "metals-colors.html"};

    private void checkPageOpened(int num) {
        assertContains(() -> getDriver().getCurrentUrl(), "/" + pages[num]);
    }

    @BeforeMethod
    public void before(final Method method) {
        isInState(SIMPLE_PAGE, method);
    }

    @Test
    public void nextTest() {
        pagination().next();
        checkPageOpened(6);
    }

    @Test
    public void prevTest() {
        pagination().previous();
        checkPageOpened(4);
    }

    @Test
    public void firstTest() {
        pagination().first();
        checkPageOpened(1);
    }

    @Test
    public void lastTest() {
        pagination().last();
        checkPageOpened(pages.length-1);
    }

    @Test
    public void jPaginationAnnotationTest() {
        Assert.assertTrue(simpleTablePage.jPagination.isDisplayed());
    }
}

File: Python/Test/jdi_uitests_webtests/test/composite/pagination_tests.py
from JDI.core.settings.jdi_settings import JDISettings
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


def check_page_opened(num):
    Assert.is_true(JDISettings.get_driver_factory().get_driver().current_url.endswith("/page{0}.htm".format(num)))


class PaginationTests(InitTests):
    pagination = EpamJDISite.simple_table_page.pagination

    def setUp(self):
        Preconditions.SIMPLE_TABLE_PAGE.is_in_state()

    def test_next(self):
        self.pagination.next()
        check_page_opened(7)

    def test_previous(self):
        self.pagination.previous()
        check_page_opened(5)

    def test_first(self):
        self.pagination.first()
        check_page_opened(1)

    def test_last(self):
        self.pagination.last()
        check_page_opened(2)